polaris/polaris/management/commands/watch_transactions.py

```diff
@@ -185,13 +185,23 @@ def find_matching_payment_op(
                 matching_payment_op = operation
                 break
 
-        return matching_payment_op
+        return PaymentOp.from_xdr(matching_payment_op.to_xdr_object().to_xdr())
 
-    @staticmethod
-    def _check_payment_op(operation: Operation, want_asset: Asset) -> bool:
+    @classmethod
+    def _check_payment_op(cls, generic_operation: Operation, want_asset: Asset) -> bool:
+        if generic_operation.TYPE != OperationType.PAYMENT:
+            return False
+        payment_operation = PaymentOp.from_xdr(
+            generic_operation.to_xdr_object().to_xdr()
+        )
+        asset_num_x = (
+            payment_operation.asset.alpha_num4 or payment_operation.asset.alpha_num12
+        )
+        asset_code = (
+            asset_num_x.asset_code.asset_code4 or asset_num_x.asset_code.asset_code12
+        ).decode()
         return (
-            operation.type_code() == Xdr.const.PAYMENT
-            and str(operation.destination) == want_asset.distribution_account
-            and str(operation.asset.code) == want_asset.code
-            and str(operation.asset.issuer) == want_asset.issuer
+            str(payment_operation.destination) == want_asset.distribution_account
```

Comment:
@overcat same here, can I just cast `operation` to get the `G...`/`M...` address? It looks like it returns an object representation:
```
class MuxedAccount:
    ...
    def __str__(self):
        out = []
        out.append(f"type={self.type}")
        out.append(f"ed25519={self.ed25519}") if self.ed25519 is not None else None
        out.append(f"med25519={self.med25519}") if self.med25519 is not None else None
        return f"<MuxedAccount {[', '.join(out)]}>"

```